pkg/operation/botanist/component/extensions/operatingsystemconfig/executor/templates/scripts/execute-cloud-config.tpl.sh

```diff
@@ -229,7 +229,11 @@ rm "$PATH_CLOUDCONFIG" "$PATH_CCD_SCRIPT_CHECKSUM"
 
 # Now that the most recent cloud-config user data was applied, let's update the checksum/cloud-config-data annotation on
 # the Node object if possible and store the current date.
-if [[ ! -z "$NODENAME" ]] && [[ -f "$PATH_CHECKSUM" ]]; then
-  {{ .pathBinaries }}/kubectl --kubeconfig="{{ .pathKubeletKubeconfigReal }}" annotate node "$NODENAME" "checksum/cloud-config-data=$(cat "$PATH_CHECKSUM")" --overwrite
+if [[ ! -z "$NODENAME" ]]; then
+  {{ .pathBinaries }}/kubectl --kubeconfig="{{ .pathKubeletKubeconfigReal }}" label node "$NODENAME" "{{ .labelWorkerKubernetesVersion }}={{ .kubernetesVersion }}" --overwrite
```

Comment:
Good catch, you're right. The values provided via `--node-labels` are not reconciled for existing `Nodes` but only added when the `Node` is initially created. When we are doing Kubernetes patch version updates then the `Node` is kept and the kubelet is restarted with the new Kubernetes version. In such case it would not update the version label, hence, we need the `cloud-config-downloader` to help out with it.